state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1816,17 +1942,37 @@ pub mod pallet {
 							},
 						};
 
-					// See if princial swap is needed, schedule it first if so:
+					// See if principal swap is needed, schedule it first if so:
 					if input_asset != output_asset && !principal_swap_amount.is_zero() {
+						let mut dca_state = DcaState {
+							scheduled_chunk_swap_id: None,
+							remaining_input_amount: principal_swap_amount,
+							// No DCA is equivalent to DCA with 1 chunk
+							remaining_chunks: dca_params.map(|p| p.number_of_chunks).unwrap_or(1),
+							accumulated_output_amount: 0,
+						};
+						let chunk_input_amount = dca_state.prepare_next_chunk();
+
+						let swap_refund_params = refund_params.as_ref().map(|params| {
+							utilities::calculate_swap_refund_parameters(
```

Comment:
Seems a bit odd to have a utility for this? 

Could it not be a method on one of the structs?

Reply:
I have a slight bias towards functions over methods, but maybe a method would work better here. Updating.